app/parser/src/log.server.ts

```diff
@@ -24,14 +23,24 @@ const stringToLevelMap: Record<string, LOG_LEVEL> = {
 
 const { ERROR, WARN, INFO, DEBUG } = LOG_LEVEL_LABEL
 
-export function getLogLevel() {
+function setIntialLogLevel() {
   if (typeof process.env.LOG_LEVEL === "string")
     return stringToLevelMap[process.env.LOG_LEVEL.toUpperCase()]
   if (typeof process.env.LOG_LEVEL === "number") return process.env.LOG_LEVEL
   return null
 }
 
-const logLevel = getLogLevel()
+let logLevel = setIntialLogLevel()
+
+export const getLogLevel = () => logLevel
+
+export function setLogLevel(level: string) {
+  const newLevel = stringToLevelMap[level.trim().toUpperCase()]
+  if(typeof newLevel === "undefined") {
```

Comment:
Is this check different from *!newLevel*?

Reply:
Yes, since silent is 0 in the `LOG_LEVEL` enum, it would prouce a falsy value when casted to a boolean and therefore, you were unable to set log level to SILENT.